Wikipedia/Code/DiffHeaderCompareView.swift

```diff
@@ -142,15 +144,32 @@ class DiffHeaderCompareView: SetupView {
     override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
         super.traitCollectionDidChange(previousTraitCollection)
 
+        setupStackView()
+
+        stackView.setNeedsLayout()
+        stackView.layoutIfNeeded()
+
+        updateFonts(with: traitCollection)
+    }
+
+    func setupStackView() {
```

Comment:
One issue exposed by this change is the metadata/button stack when there's no change message is present doesn't render correctly. I'd expect perhaps that the stack should top aligned in this case, instead of filling in extra space to match the height of its sibling stack, but this may require design input to confirm what the intention should be.

![stack_spacing](https://github.com/wikimedia/wikipedia-ios/assets/5652327/cb659591-1c2c-46b4-9bc1-72ce9fdbfa44)

Reply:
I'll get Robin's feedback on this case before updating

Reply:
This one will be handled in a separate ticket: https://phabricator.wikimedia.org/T347425